### PR TITLE
tests(iroh): Feature-flag the doc test

### LINKED ISSUE #2950: iroh-net: default features are not compatible with doc tests
```
$ cargo test -p iroh-net
```

fails with 
```
---- iroh-net/src/discovery.rs - discovery (line 73) stdout ----
error[E0432]: unresolved import `iroh_net::discovery::local_swarm_discovery`
   --> iroh-net/src/discovery.rs:76:26
    |
6   | use iroh_net::discovery::local_swarm_discovery::LocalSwarmDiscovery;
    |                          ^^^^^^^^^^^^^^^^^^^^^ could not find `local_swarm_discovery` in `discovery`
    |
note: found an item that was configured out
   --> /Users/dignified/opensource/iroh/iroh-net/src/discovery.rs:115:9
    |
115 | pub mod local_swarm_discovery;
    |         ^^^^^^^^^^^^^^^^^^^^^
note: the item is gated behind the `discovery-local-network` feature
   --> /Users/dignified/opensource/iroh/iroh-net/src/discovery.rs:113:7
    |
113 | #[cfg(feature = "discovery-local-network")]
    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0432`.
Couldn't compile the test.

failures:
    iroh-net/src/discovery.rs - discovery (line 73)
```

### PR DESCRIPTION
**Pull request by flub**
## Description

This test can only work if the feature is enabled.  Feature-flag the
entire test contents.

Closes #2950.

## Change checklist

- [x] Self-review.
- [x] Documentation updates following the [style guide](https://rust-lang.github.io/rfcs/1574-more-api-documentation-conventions.html#appendix-a-full-conventions-text), if relevant.
- [x] Tests if relevant.
- [x] All breaking changes documented.

### CODE REVIEW

**Comment by dignifiedquire** [comment]
> To also enable [`LocalSwarmDiscovery`], it can be added as another service in the

will still fail in `cargo doc` as the link is invalid with default features

**Comment by dignifiedquire** [comment]
> Do we support cargo doc without all features?

I think we should, because locally the initial thing you do is run `cargo doc`

> Should we not make sure CI fails on these things?

we probably should

**Comment by flub** [comment]
We have
```
[package.metadata.docs.rs]
all-features = true
```
So I think everything will work fine.  Of course you can argue this is not a great local experience.  But neither is having some of the docs accidentally hidden when building locally.  Not sure what to do here...

**Comment by dignifiedquire** [comment]
> But neither is having some of the docs accidentally hidden when building locally

no need to hide things, just don't make it a link

**Review by dignifiedquire — APPROVED** [review]

File: iroh/src/discovery.rs [hunk 1]
```diff
@@ -67,10 +69,18 @@
 //! # }
 //! ```
 //!
-//! To also enable [`LocalSwarmDiscovery`], it can be added as another service in the
+//! To also enable
+#![cfg_attr(feature = "discovery-local-network", doc = "[`LocalSwarmDiscovery`]")]
+#![cfg_attr(
+    not(feature = "discovery-local-network"),
+    doc = "`LocalSwarmDiscovery`"
+)]
```
[the remark this answers is not in the document]
Comment (flub) [reply iroh/src/discovery.rs:77]: You asked for this :stuck_out_tongue_closed_eyes:

**Pull request merged**